src/components/Form.js

```diff
@@ -149,12 +150,30 @@ export default class Form extends Component {
     }
   };
 
+  getUsedFormData = event => {
+    const fields = event.target.elements;
```

Comment:
@travisdahl @CodeGains It seems like `event.target.elements` will only return [the following elements](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/elements): `<button> <fieldset> <input> <object> <output> <select> <textarea>`.

How would we include custom widgets that don't render any of these tags?

I think that, instead of this approach, we may have to go with using ajv's [`removeAdditional`](https://github.com/epoberezkin/ajv#filtering-data) method.

Reply:
I'll have to look more into the custom widgets. What kind of a widget would not have any of these elements? 

Perhaps we also need to add an additional prop for `name` for the custom widgets. The current custom widgets props for reference: https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/#custom-widgets-and-fields

Reply:
Good point about the `name` prop.

Regarding a custom widget without any of those elements off the top of my head, but one could imagine, say, using a [meter](https://www.w3schools.com/tags/tag_meter.asp) element that could be controlled by buttons.

Or, maybe one creates a custom widget for money with a div that contains two textboxes -- one which controls the number of dollars and the other which controls the number of cents -- and the "div" element itself will report out the value.

Basically, the way a custom widget is designed, we currently don't impose the restriction that elements `<button> <fieldset> <input> <object> <output> <select> <textarea>` have to report out the value; rather, we let it be flexible by using the `onChange` prop. Ideally, we would keep it that way.

Reply:
I see your theoretical point, BUT i would say that is bad practice to not set some kind of input value, even if its a hidden input.  No matter what your visual UI is, if its a form element, it should set a value somewhere, for example if i made any of the custom widgets you suggested..

• a meter with buttons. On up and down button clicks, maybe a graphical ui shows a meter at 50%, but I would set a hidden input value to `50` and that input would be tracked. 

• in the money example, same thing, your dollar and cents can be 2 fields without an onchange but they should be setting a caoncatted value in a hidden input that is being tracked


Thats said, I have an idea on how to support all elements like you are asking, but they would still need to provide a `name` attribute to that element. or maybe we use something like `data-path`. I just like the name attribute because its a more standardized, traditional pattern on how to use forms.

Reply:
@travisdahl here's a more realistic example: using a `<div contenteditable>`

https://stackoverflow.com/questions/22677931/react-js-onchange-event-for-contenteditable

Reply:
ah yes, I see your point. That is definitely a valid use case.

Reply:
I think ajv's [`removeAdditional`](https://github.com/epoberezkin/ajv#filtering-data) method would work well to overcome this problem though?

Reply:
If we used `removeAdditional` wouldn't we need the schema with the dependent fields removed if they weren't being used?

Reply:
@CodeGains can you elaborate on what you mean here?

Reply:
Unless I am misunderstanding how `removeAdditional` works, you would need the schema in order to see what fields in the formData are 'additional'. But in the case like #754, `billing_address` still exists when `credit_card` is removed. However, `billing_address` will always be in the schema, so `removeAdditional` will not remove `billing_address` from the formData.